packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -499,8 +583,11 @@ export class DataModel<
         if (groupProcessors.length > 0) {
             this.postProcessGroups(processedData);
         }
-        if (aggregates.length > 0) {
-            this.aggregateData(processedData);
+        if (aggregates.length > 0 && processedData.type === 'ungrouped') {
+            this.aggregateUngroupedData(processedData);
+        }
+        if (aggregates.length > 0 && processedData.type === 'grouped') {
```

Comment:
nit: `else if` to reduce an unnecessary check

Reply:
Done